### PR TITLE
Bug/register interfaces to types regression

### LINKED ISSUE #1957: Regression: register_graphql_interfaces_to_types() not working
With the release of WPGraphQL v1.4.0, the `register_graphql_interfaces_to_types()` method is no longer working. 

This was reported by @chriszarate in Slack. 

It appears to be a regression caused by the `graphql_object_type_interfaces` being removed.

### PR DESCRIPTION
**Pull request by jasonbahl**
What does this implement/fix? Explain your changes.
---------------------------------------------------

- fixes regression to the `register_graphql_interfaces_to_types()` method 
- renames filter `graphql_object_type_interfaces` to `graphql_type_interfaces`. If you were using `graphql_object_type_interfaces` filter in your codebase, you will need to update the filter name as it now applies to both Object and Interface Types.
- applies `graphql_type_interfaces` to both WPInterfaceType and WPObjectType classes
- adds tests for registering interfaces to types and registering interfaces to other interfaces.

Does this close any currently open issues?
------------------------------------------
closes #1957

### CODE REVIEW

**Review by chriszarate — APPROVED** [review]
Gave this branch a spin. Filter is called correctly and was I able to add interfaces to existing types via the `graphql_object_type_interfaces` filter. I didn't test the `wp-graphql-gutenberg` plugin but I would expect it to work fine.

Thank you!

**Comment by shrenikTrukky** [comment]
wp-graphql-gutenberg 's query for blocks not working with new version of wp-graphql 1.4.0.

**Comment by jasonbahl** [comment]
@chriszarate @shrenikTrukky thanks for testing and confirming! Apologies for this regression. We have better tests in place now to prevent this in the future.

File: .github/workflows/testing-integration.yml [hunk 1]
```diff
@@ -64,6 +64,7 @@ jobs:
           DEBUG: ${{ matrix.debug }}
           SKIP_TESTS_CLEANUP: ${{ matrix.coverage }}
           SUITES: wpunit
+          COMPOSER_MEMORY_LIMIT: "-1"
```
Comment (chriszarate) [inline .github/workflows/testing-integration.yml:67]: ♾️

**Review by chriszarate — APPROVED** [review]
Tested again with latest changes, looks good and filter works as expected. Thanks!

File: .github/workflows/testing-integration.yml (hunk 1 above)
Comment (jasonbahl) [reply .github/workflows/testing-integration.yml:67]: ya, trying to overcome some memory limit issues, but going to make that a separate issue

**Review by chriszarate — APPROVED** [review]

**Pull request merged**